model_signing/signing/in_toto.py

```diff
@@ -805,3 +810,49 @@ def manifest_from_payload(
             items.append(item)
 
         return manifest_module.ShardLevelManifest(items)
+
+
+class IntotoSignature(signing.Signature):
+
+    def __init__(self, bundle: bundle_pb.Bundle):
+        self._bundle = bundle
+
+    @override
+    def write(self, path: pathlib.Path) -> None:
+        path.write_text(self._bundle.to_json())
+
+    @classmethod
+    @override
+    def read(cls, path: pathlib.Path) -> Self:
+        bundle = bundle_pb.Bundle().from_json(path.read_text())
+        return cls(bundle)
+
+    def to_manifest(self) -> manifest_module.Manifest:
+        payload = json.loads(self._bundle.dsse_envelope.payload)
+        return IntotoPayload.manifest_from_payload(payload)
```

Comment:
I'm using the existing code from the signature package. The implementation in signing/sigtore uses the Bundle class instead of the proto which makes it more complicated to use with other signature providers that already exist.

Reply:
We want to use `sigstore-python`'s `Bundle` class as it has validation for invalid bundles. We don't want to neither reimplement those validations in our code nor to skip those validations

Reply:
We want to support other signing methods than sigstore but still use the bundle format and therefore we need to use the proto because the bundle class only provides a classmethod from json. Does it make sense to first create the proto, serialize it to json just for the bundle class to parse it back? 

Furthermore, the checks in the bundle class fail if there is anything else then a cert chain.

Reply:
> Furthermore, the checks in the bundle class fail if there is anything else then a cert chain.

Can you expand on this please?

Reply:
For the sigstore bundle class the bundle must contain a cert chain, see the _verify method (https://github.com/sigstore/sigstore-python/blob/main/sigstore/models.py#L379). That means we cannot use it for storing just a key in the verification material because that check would just fail and reject a perfectly valid sigstore bundle. It makes sense for it to fail because the class is part of sigstore and therefore applies checks to only verify it with sigstore.
It also fails if the bundle does not contain log entries.

Reply:
Makes sense. I'd still want to do the right thing here in the long term and have validations (we could probably open a feature request on sigstore-python to support these use cases). But also don't want to delay you.

Can you move these classes to `signature/` instead? It helps with moving forward without delay and also with layering, since they rely on `signing/` code

Reply:
Could you please explain why? 
They actually follow the base class layout as defined by you and provide functionality for intoto based payloads. IMO we should keep the classes in the in_toto module as they fit there.

Reply:
The other reason is that now this file depends on sigstore, but internally we would need to import a lot of libraries to use this import. So we cannot experiment internally if the changes come here (unless we remove the code via copybara, but that's equivalent to it being in a separate place anyway)

Reply:
That's not experimental we want an in_toto signer that supports different signing mechanisms. We still need to define where we merge the signing mechanisms though. We should add the in_toto parts where they are right now. The rest isn't as stable as you make it sound anyway with all the tweaks that happened in the last big pr.

Reply:
The only changes in #276 in this file were adding a new method to every class and fixing a bug to have the name field in every subject.

The issue that we'd have to import too many packages due to the sigstore import is still there.

Reply:
Ok, then use copybara the outside community can still get the benefit from using it as it is, i.e. utilizing a single class to support 3 different signing and verification methods.

Reply:
I'm using copybara. It's flaky if you ask it to remove text from files, so I'd prefer to use it to remove paths completely.